src/sentry/rules/processing/delayed_processing.py

```diff
@@ -200,6 +202,31 @@ def parse_rulegroup_to_event_data(
     return parsed_rulegroup_to_event_data
 
 
+def bulk_fetch_events(event_ids: list[str], project_id: int) -> set[Event]:
```

Comment:
This was largely copied from https://github.com/getsentry/getsentry/blob/b2689f4e98cb5d641ecdda4605e9cf84932b2a67/bin/backfill_perf_issue_events_issue_platform.py#L506-L530 - but I removed the `time.sleep`s. It seems like this could be useful somewhere else, I was surprised we didn't already have this function, though I'm not sure where would be a good place for it to live.

Reply:
🤔 If I were looking for this code I'd probably check in `sentry.eventstore.models` where the `Event` is imported from. I'm thinking basically i know what model i want, and then i can either get 1 or many of them from there.

Reply:
I think for _right now_ I wanna keep it where it is, then when I have more confidence it's working as expected it can be moved for others to use.